integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go

```diff
@@ -0,0 +1,127 @@
+	require.Nil(t, err)
+	require.NotNil(t, stakeTx)
+
+	delegationAddress := stakeTx.Logs.Events[2].Address
+	sp1AddressBytes, _ := cs.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(delegationAddress)
+
+	txStake = chainSimulatorIntegrationTests.GenerateTransaction(sp2Owner.Bytes, staking.GetNonce(t, cs, sp2Owner), vm.DelegationManagerSCAddress, stakeValue, dataField, 80_000_000)
+	stakeTx, err = cs.SendTxAndGenerateBlockTilTxIsExecuted(txStake, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, stakeTx)
+
+	delegationAddress = stakeTx.Logs.Events[2].Address
+	sp2AddressBytes, _ := cs.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(delegationAddress)
+
+	fmt.Println(sp1AddressBytes, sp2AddressBytes)
+
+	// do delegate to sp1
+	delegateValue := big.NewInt(0).Mul(big.NewInt(500), chainSimulatorIntegrationTests.OneEGLD)
+	txDelegate1 := chainSimulatorIntegrationTests.GenerateTransaction(delegator.Bytes, 0, sp1AddressBytes, delegateValue, "delegate", gasLimitForDelegate)
+	delegate1Tx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txDelegate1, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, delegate1Tx)
+
+	err = cs.ChangeEpochs(40)
+	require.Nil(t, err)
+
+	// move 100 egld from sp1 --> sp2 should work
+	moveValue := big.NewInt(0).Mul(big.NewInt(100), chainSimulatorIntegrationTests.OneEGLD)
+	dataField = fmt.Sprintf("changeStakingProvider@%s@%s@%s", hex.EncodeToString(moveValue.Bytes()), hex.EncodeToString(sp1AddressBytes), hex.EncodeToString(sp2AddressBytes))
+	tx := chainSimulatorIntegrationTests.GenerateTransaction(delegator.Bytes, staking.GetNonce(t, cs, delegator), vm.DelegationManagerSCAddress, big.NewInt(0), dataField, 80_000_000)
+	moveDelegationTx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, moveDelegationTx)
+
+	// move 100 egld from sp2 --> sp1 should NOT work (is in cool down period)
+	moveValue = big.NewInt(0).Mul(big.NewInt(100), chainSimulatorIntegrationTests.OneEGLD)
+	dataField = fmt.Sprintf("changeStakingProvider@%s@%s@%s", hex.EncodeToString(moveValue.Bytes()), hex.EncodeToString(sp2AddressBytes), hex.EncodeToString(sp1AddressBytes))
+	tx = chainSimulatorIntegrationTests.GenerateTransaction(delegator.Bytes, staking.GetNonce(t, cs, delegator), vm.DelegationManagerSCAddress, big.NewInt(0), dataField, 80_000_000)
+	moveDelegationTx, err = cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, moveDelegationTx)
+
+	eventWithError := moveDelegationTx.Logs.Events[0]
+	require.Equal(t, eventWithError.Identifier, core.SignalErrorOperation)
+	require.Equal(t, "cannot migrate to another service provider during cooldown period, wait 30 more epochs", string(eventWithError.Topics[1]))
+
+	// check delegation values for delegator in sp1 and sp2
```

Comment:
can you please add a test for moving another 100 EGLD from sp1 to sp2. Should it work or not? I'm not sure. 
How about moving 100 EGLD from sp1 to sp3? 
Maybe there are unit tests for these cases...

Reply:
This case will work, you will have the cool-down period on the destination 
You will can migrate again delegation from sp2 only after cool-down period

Reply:
ok